teuthology/test/test_misc.py

```diff
@@ -150,3 +150,15 @@ def test_missing_b_deep_merge(self):
     def test_invalid_b_deep_merge(self):
         with pytest.raises(AssertionError):
             misc.deep_merge({"a": "b"}, "invalid")
+
+
+class TestIsInDict(object):
+    def test_is_in_dict(self):
+        assert misc.is_in_dict('a', 'foo', {'a':'foo', 'b':'bar'})
+        assert misc.is_in_dict(
+            'a', {'sub1':'key1', 'sub2':'key2'},
+            {'a':{'sub1':'key1', 'sub2':'key2', 'sub3':'key3'}}
+        )
+
+        assert misc.is_in_dict('a', 'foo', {'a':'bar', 'b':'foo'}) == False
+        assert misc.is_in_dict('a', 'foo', {'a':{'a': 'foo'}}) == False
```

Comment:
Nitpick, but linters will complain: with booleans, it's more Pythonic to test for ``if foo is False`` rather than use ``==``